src/software/jetson_nano/services/motor.cpp

```diff
@@ -93,9 +125,137 @@ std::unique_ptr<TbotsProto::DriveUnitStatus> MotorService::poll(
     return std::make_unique<TbotsProto::DriveUnitStatus>();
 }
 
+void MotorService::spiTransfer(int fd, uint8_t const* tx, uint8_t const* rx, unsigned len)
+{
+    int ret;
+
+    struct spi_ioc_transfer tr[1];
+    memset(tr, 0, sizeof(tr));
+
+    tr[0].tx_buf        = (unsigned long)tx;
+    tr[0].rx_buf        = (unsigned long)rx;
+    tr[0].len           = len;
+    tr[0].delay_usecs   = 0;
+    tr[0].speed_hz      = SPI_SPEED_HZ;
+    tr[0].bits_per_word = 8;
+
+    ret = ioctl(fd, SPI_IOC_MESSAGE(1), &tr);
+
+    if (ret < 1)
+    {
+        LOG(FATAL) << "SPI Transfer to motor failed, not safe to proceed";
```

Comment:
for system calls like this is it worth also logging the kind of error raised

Reply:
added the errno log